src/ketrew_backend/plugin.ml

```diff
@@ -0,0 +1,167 @@
+
+open Coclobas
+open Internal_pervasives
+
+let name = "coclobas-kube"
+
+module Run_parameters = struct
+
+  type t = [
+    | `Created of Client.t * Kube_job.Specification.t
+    | `Running of Client.t * Kube_job.Specification.t
+  ] [@@deriving yojson, show]
+
+  let serialize run_parameters =
+    to_yojson run_parameters
+    |> Yojson.Safe.pretty_to_string ~std:true
+
+  let deserialize_exn s =
+    Yojson.Safe.from_string s |> of_yojson
+    |> function
+    | `Ok o -> o
+    | `Error e -> failwith e
+end
+
+let create ~base_url spec = 
+  `Long_running (name, `Created (Client.make base_url, spec) |> Run_parameters.serialize)
+
+module Long_running_implementation : Ketrew.Long_running.LONG_RUNNING = struct
+
+  type run_parameters = Run_parameters.t
+  include Run_parameters
+     
+  let name = "coclobas-kube"
+
+  module KLRU = Ketrew.Long_running_utilities
+
+  let serialize run_parameters =
+    Run_parameters.to_yojson run_parameters
+    |> Yojson.Safe.pretty_to_string ~std:true
+
+  let deserialize_exn s =
+    Yojson.Safe.from_string s |> Run_parameters.of_yojson
+    |> function
+    | `Ok o -> o
+    | `Error e -> failwith e
+
+  let classify_client_error m =
+    m >>< function
+    | `Ok o -> return o
+    | `Error (`Client cl) -> fail (`Fatal (Client.Error.to_string cl))
+    | `Error (`Coclo_plugin (`Expecting_one_status l)) ->
+      fail (`Fatal (sprintf "Expecting one status but got: [%s]"
+                      (List.map l ~f:(fun (id, st) -> id)
+                       |> String.concat ~sep:"; ")))
+
+  let start :
+    run_parameters ->
+    host_io:Ketrew.Host_io.t ->
+    (run_parameters, Ketrew.Host_io.Error.classified) Deferred_result.t =
+    fun rp ~host_io ->
+      match rp with
+      | `Running _ ->
+        ksprintf KLRU.fail_fatal "start on already running: %s"
+          (Run_parameters.show rp)
+      | `Created (client, spec) ->
+        classify_client_error begin
+          Client.submit_kube_job client spec
+          >>= fun () ->
+          return (`Running (client, spec))
+        end
+
+  let running rp f =
+    match rp with
+    | `Created _ ->
+      ksprintf KLRU.fail_fatal "update on not started: %s"
```

Comment:
I am probably not understanding what's going on here. It seems concievable that Ketrew would submit a job, which would be in the "Created" state, and before it could transition to "running", a Ketrew user might want to kill the job. Then you'd have this error, no?